File: Behavioral/Observer.py
class JobPost:
    _title = None

    def __init__(self, title):
        self.title = title

    def getTitle(self):
        return self.title

class JobSeeker:
    _name = None

    def __init__(self, name):
        self.name = name

    def onJobPosted(self, job):
        print 'Hi ' + self.name + '! New job posted: ' + job.getTitle()

class EmploymentAgency:
    _observers = []

    def notify(self, jobPosting):
        for observer in self._observers:
            observer.onJobPosted(jobPosting)

    def attach(self, observer):
        self._observers.append(observer)

    def addJob(self, jobPosting):
        self.notify(jobPosting)


johnDoe = JobSeeker('John Doe')
janeDoe = JobSeeker('Jane Doe')

jobPostings = EmploymentAgency()
jobPostings.attach(janeDoe)
jobPostings.attach(johnDoe)

jobPostings.addJob(JobPost('Software Engineer'))
'''
Output
Hi John Doe! New job posted: Software Engineer
Hi Jane Doe! New job posted: Software Engineer
'''

File: examples-per-language/php/behavioral/Observer.php
<?php

use designPatternsForHumans\behavioral\Observer\JobPost;
use designPatternsForHumans\behavioral\Observer\JobPostings;
use designPatternsForHumans\behavioral\Observer\JobSeeker;

require_once __DIR__ . '/autoload.php';

$john = new JobSeeker('John Doe');
$jane = new JobSeeker('Jane Doe');

$jobPostings = new JobPostings();
$jobPostings->attach($john);
$jobPostings->attach($jane);

$jobPostings->add(new JobPost('Software Engineer'));
